services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -2093,6 +2094,255 @@ Environment "acceptance-de" has both upstream.latest and upstream.environment co
 	}
 }
 
+func TestReleaseTrainWithCommit(t *testing.T) {
+	tcs := []struct {
+		Name               string
+		SetupTransformers  []Transformer
+		ReleaseTrainEnv    string
+		expectedError      string
+		expectedCommitMsg  string
+		shouldSucceed      bool
+		overrideCommitHash string
+	}{
+		{
+			Name: "Release train done without commit Hash",
+			SetupTransformers: []Transformer{
+				&CreateEnvironment{
+					Environment: "dev",
+					Config:      testutil.MakeEnvConfigLatest(nil),
+				},
+				&CreateApplicationVersion{
+					Application: "test",
+					Version:     1,
+					Manifests: map[string]string{
+						"dev":     "dev",
+						"staging": "staging",
+					},
+				},
+				&CreateEnvironment{
+					Environment: "staging",
+					Config: config.EnvironmentConfig{
+						Upstream: &config.EnvironmentConfigUpstream{
+							Latest:      false,
+							Environment: "dev",
+						},
+					},
+				},
+			},
+			ReleaseTrainEnv:    "staging",
+			shouldSucceed:      true,
+			expectedError:      "",
+			expectedCommitMsg:  "Release Train to environment/environment group 'staging':\n\nRelease Train to 'staging' environment for 1 apps:\n\nThe release train deployed 1 services from 'dev' to 'staging'\ndeployed version 1 of \"test\" to \"staging\"\n\n\n\n\n",
+			overrideCommitHash: "",
+		},
+		{
+			Name: "Release train done with commit Hash",
```

Comment:
I don't see a commit hash in this test, please rename it.

Reply:
The commit hash has to be generated once the setupTransformers are run here: https://github.com/freiheit-com/kuberpult/pull/1358/files#diff-894f09b2c5166abb03bb01b74cfbe3a0c17322a4fe3a938f2acf819d40bd3269R2310